### PR TITLE
AMP-66640 [streaming] Update identify forwarding description

### PR DESCRIPTION
**Pull request by sina-jamshidi**
# Amplitude Developer Docs PR
We are adjusting user forwarding feature so updating the descriptions

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

File: docs/data/destinations/braze.md [hunk 1]
```diff
@@ -45,10 +45,10 @@ After you create the destination, you must configure the settings.
 2. Select your **Braze REST API Endpoint**.
 3. Enter your **REST API Key**.
 4. Enter your **Braze App Identifier API Key**.
-5. **Create & Update Users** creates users in Braze and update the properties of existing users when an Amplitude Identify API call is made.
+5. **Create & Update Users** creates users in Braze and update the properties of existing users when a user is created or user properties are updated in Amplitude.
```
Comment (sina-jamshidi) [inline docs/data/destinations/braze.md:48]: Sanity check. Is sentence this confusing?

Comment (caseyamp) [reply docs/data/destinations/braze.md:48]: A little. How does an existing user get created? Does this mean creates or (inclusive `or`) updates users in Braze when their corresponding user is created or updated in Amplitude?

```suggestion
5. Configure **Create & Update Users**. This setting does two things: creates users in Braze when they're created in Amplitude and updates existing users when their user properties are updated in Amplitude.
```

File: docs/data/destinations/braze.md [hunk 2]
```diff
@@ -45,10 +45,10 @@ After you create the destination, you must configure the settings.
...
       1. To create and update users, toggle **Create & Update Users** to **Enabled**.
       2. To select user properties to send, expand the **Specify user properties to send** panel, and select properties to forward. If you don't select any properties here, Amplitude doesn't include any.
-      3. Check the **Update Users Only** box to not create users if they don't exist when forwarding Identify API calls. If you're sending alias-only user profiles, this field shouldn't be checked. If **Update Users Only** is checked, your alias-only user profiles won't be created in Braze. More information available in the [Braze API documentation](https://www.braze.com/docs/api/objects_filters/user_attributes_object).
+      3. Check the **Update Users Only** box to not create users if they don't exist in Braze. If you're sending alias-only user profiles, this field shouldn't be checked. If **Update Users Only** is checked, your alias-only user profiles won't be created in Braze. More information available in the [Braze API documentation](https://www.braze.com/docs/api/objects_filters/user_attributes_object).
 6. **Send Events** sends events ingested by Amplitude to Braze.
```
Comment (caseyamp) [inline docs/data/destinations/braze.md:52]:
```suggestion
6. Configure **Send Events** to send events ingested by Amplitude to Braze.
```

Sorry for the out of scope suggestion, but I think we have an opportunity to update for some of the changes you're making here -- all the other "steps" are verbs, and it's weird that I just wrote setting names here.

File: docs/data/destinations/braze.md (hunk 1 above)
Comment (caseyamp) [reply docs/data/destinations/braze.md:48]: @sina-jamshidi I made a suggestion that hopefully makes this more clear in the docs.

**Review by Brandoncharleskhoo1 — APPROVED** [review]
Thanks for driving this team! Looks good

**Pull request merged**